### PR TITLE
gui: lvgl: Set actual display size when using static allocated rendering buffers

### PR DESCRIPTION
**Pull request by vanwinkeljan**
Set actual display size, obtained via the display driver, when using static allocated rendering buffers.
If the actual screen size is not set an out-of-bound write could occur in case the maximum resolution settings
for LVGL are larger than the actual screen resolution.

### CODE REVIEW

File: lib/gui/lvgl/lvgl.c [hunk 1]
```diff
@@ -69,6 +69,19 @@ static void lvgl_log(lv_log_level_t level, const char *file, uint32_t line,
...
 {
+	struct display_capabilities cap;
+	struct device *display_dev = (struct device *)disp_drv->user_data;
+
+	display_get_capabilities(display_dev, &cap);
+
+	__ASSERT(cap.x_resolution <= CONFIG_LVGL_HOR_RES,
+		 "Horizontal resolution is larger then maximum");
+	__ASSERT(cap.y_resolution <= CONFIG_LVGL_VER_RES,
+		 "Vertical resolution is larger then maximum");
```
Comment (gmarull) [inline lib/gui/lvgl/lvgl.c:80]: `then` -> `than`

File: lib/gui/lvgl/lvgl.c [hunk 2]
```diff
@@ -69,6 +69,19 @@ static void lvgl_log(lv_log_level_t level, const char *file, uint32_t line,
 
 static int lvgl_allocate_rendering_buffers(lv_disp_drv_t *disp_drv)
 {
+	struct display_capabilities cap;
+	struct device *display_dev = (struct device *)disp_drv->user_data;
+
+	display_get_capabilities(display_dev, &cap);
+
+	__ASSERT(cap.x_resolution <= CONFIG_LVGL_HOR_RES,
```
Comment (jfischer-phytec-iot) [inline lib/gui/lvgl/lvgl.c:77]: I suggest to treat it as error and not just ASSERT which is not enabled per default.

File: lib/gui/lvgl/lvgl.c [hunk 3]
```diff
@@ -69,6 +69,19 @@ static void lvgl_log(lv_log_level_t level, const char *file, uint32_t line,
...
+		 "Horizontal resolution is larger then maximum");
```
Comment (gmarull) [inline lib/gui/lvgl/lvgl.c:78]: `then` -> `than`

File: drivers/display/display_dummy.c [hunk 4]
```diff
@@ -29,6 +29,15 @@ static int dummy_display_write(const struct device *dev, const uint16_t x,
...
 {
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller then width");
+	__ASSERT(desc->pitch <= CONFIG_DUMMY_DISPLAY_X_RES,
+		"Pitch in descriptor is larger than screen size");
+	__ASSERT(desc->height <= CONFIG_DUMMY_DISPLAY_Y_RES,
+		"Height in descriptor is larger than screen size");
+	__ASSERT(x + desc->pitch <= CONFIG_DUMMY_DISPLAY_X_RES,
+		 "Writing outside screen boundaries in horizontal direction");
+	__ASSERT(y + desc->height <= CONFIG_DUMMY_DISPLAY_Y_RES,
+		 "Writing outside screen boundaries in vertical direction");
```
Comment (gmarull) [inline drivers/display/display_dummy.c:40]: I think that display API should clarify what runtime conditions are allowed/forbidden. E.g. when writing outside boundaries, should it be an assertion or an error return (`EINVAL, ERANGE`)? All drivers should exhibit the same behavior.

File: lib/gui/lvgl/lvgl.c (hunk 2 above)
Comment (gmarull) [inline lib/gui/lvgl/lvgl.c:77]: These assertions could be turned into build time assertions if all displays use DT (not for now, just suggesting)

File: lib/gui/lvgl/lvgl.c (hunk 1 above)
Comment (gmarull) [inline lib/gui/lvgl/lvgl.c:80]: `then` -> `than`

File: drivers/display/display_dummy.c (hunk 4 above)
Comment (vanwinkeljan) [reply drivers/display/display_dummy.c:40]: Not sure if it needs to be enforced by the API, I think it is up to each driver to handle this case. Could well be that display controller has no issue with out of bound writes.

On the other hand as this is a test driver I will return `ERANGE` next to the assertion.

File: lib/gui/lvgl/lvgl.c (hunk 2 above)
Comment (vanwinkeljan) [reply lib/gui/lvgl/lvgl.c:77]: Start to wonder if this should be handled as an assert and/or error at all.
In the end there is no issue with only partially using the display.

**Review by gmarull — APPROVED** [review]

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**